### PR TITLE
add Junit output formatter

### PR DESCRIPTION
**Pull request by PleaseHelpMe-hub**

### CODE REVIEW

**Comment by DaveLiddament** [comment]
@PleaseHelpMe-hub Thanks for this PR, junit output format is really useful, so thanks for this. 

I'll look at it in more detail over the next couple of days.

File: src/Plugins/OutputFormatters/JunitOutputFormatter.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        $test['tests'] = $testCount;
+        $test['failures'] = $testCount;
+
+        $suitCount = 0;
+        $caseCount = 0;
+        $oldRel = null;
+        $testsuite = null;
+        foreach ($analysisResults->getAnalysisResults() as $analysisResult) {
+            $details = $analysisResult->getFullDetails();
+            $type = $details['type'] ?? 'error';
```
Comment (DaveLiddament) [inline src/Plugins/OutputFormatters/JunitOutputFormatter.php:47]: What is `type` ? Should this be `$analysisResult->getType()->getType()`?

File: src/Plugins/OutputFormatters/JunitOutputFormatter.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            $type = strtolower($type);
+            $column = $details['column'] ?? '0';
```
Comment (DaveLiddament) [inline src/Plugins/OutputFormatters/JunitOutputFormatter.php:49]: What is column?

**Review by DaveLiddament — COMMENTED** [review]
Question about `$type`

File: src/Plugins/OutputFormatters/JunitOutputFormatter.php (hunk 1 above)
Comment (PleaseHelpMe-hub) [reply src/Plugins/OutputFormatters/JunitOutputFormatter.php:47]: Hey @DaveLiddament no, not really. The type we want is the Type that has the value "Warning" or "error", and if it's empty, we thought to just default it to error. Firstly, we used the getType, and it gave another value.
In my understanding, and testing we did `$analysisResult->getType()->getType() !== $details['type']`
Let me know if we unterstood something wrong here.
If you have further questioned we can discuss it of course.

**Comment by DaveLiddament** [comment]
@PleaseHelpMe-hub  thanks for you work on this. I took you starting point and made a few updates to satisfy the static analysis tools see PR #100.

**Pull request closed without merging**